exo-setup/src/docker-setup.ls

```diff
@@ -50,42 +50,48 @@ class DockerSetup
   _get-docker-env-vars: ->
     env-vars =
       ROLE: @role
-      EXOCOM_HOST: 'exocom'
-      EXOCOM_PORT: '$EXOCOM_PORT'
+    for dependency-config in @app-config.dependencies
+      dependency = ApplicationDependency.build dependency-config
+      env-vars = {...env-vars, ...dependency.get-service-env-variables!}
     for dependency of @service-config.dependencies
       env-vars[dependency.to-upper-case!] = dependency
     env-vars
 
 
   # compiles list of names of dependencies a service relies on
   _get-service-dependencies: ->
-    dependencies = ['exocom']
+    dependencies = []
+    for dependency-config in @app-config.dependencies
+      dependencies.push "#{dependency-config.type}#{dependency-config.version}"
     for dependency, dependency-config of @service-config.dependencies
-      dependencies.push (dependency + dependency-config.dev.version)
+      dependencies.push "#{dependency}#{dependency-config.dev.version}"
     dependencies
 
 
   # builds the Docker config for a service dependency
   _get-service-dependency-docker-config: (dependency-name, dependency-config) ->
     if dependency-config.volumes
-      data-path = global-exosphere-directory @app-name, dependency-name 
+      data-path = global-exosphere-directory @app-config.name, dependency-name
       fs.ensure-dir-sync data-path
       rendered-volumes =  map ((volume) -> Handlebars.compile(volume)({"EXO_DATA_PATH": data-path})), dependency-config.volumes
 
     Obj.compact do
       image: "#{dependency-config.image}:#{dependency-config.version}"
       container_name: dependency-name + dependency-config.version
-      ports: dependency-config.ports 
+      ports: dependency-config.ports
       volumes: rendered-volumes or undefined
-    
+
 
   _get-external-service-docker-config: ->
     | !@docker-image => throw new Error red "No location or docker-image specified"
+    dependencies = []
+    for dependency-config in @app-config.dependencies
+      dependencies.push "#{dependency-config.type}#{dependency-config.version}"
     docker-config = {}
     docker-config[@role] =
       image: @docker-image
       container_name: @role
-      depends_on: ['exocom']
+      depends_on: dependencies 
```

Comment:
Also does this need the env variables as well? (or anything else in `_get-service-docker-config`?